middleware.ts

```diff
@@ -0,0 +1,17 @@
+import { NextResponse, type NextRequest } from "next/server";
+
+// This function can be marked `async` if using `await` inside
+export async function middleware(request: NextRequest) {
+	if (request.nextUrl.pathname.startsWith("/cart")) {
+		const userId = request.cookies.get("access-token");
+		if (!userId) {
+			console.log("Redirect from middleware");
+			return NextResponse.rewrite(new URL("/login", request.url));
```

Comment:
mozesz dodatkowo przekazać query z info z jakiego urla poszedł redirect na login, żeby po udanym logowaniu wróciło tam skąd przyszło

Reply:
Dzięki, już na to patrzę